src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (tail != null) {
+            tail.next = node;
+        }
+        tail = node;
+        if (head == null) {
+            head = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkAddedPosition(index);
+        if (index == 0) {
+            addFirst(value);
+            return;
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
```

Comment:
why not using `else if` and `else`?

you would avoid these unnecessary `return` statements